src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,49 @@
 package core.stack;
 
+import java.util.Arrays;
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int INITIAL_CAPACITY = 10;
+    private static final int CAPACITY_MULTIPLIER = 2;
+    private Object[] stack;
+    private int size;
 
+    public MateStack() {
+        stack = new Object[INITIAL_CAPACITY];
+    }
+
+    public void push(T value) {
+        if (size == stack.length) {
+            growStack();
+        }
+        stack[size] = value;
+        size++;
     }
 
     public T peek() {
-        return null;
+        checkStack();
+        return (T) stack[size - 1];
     }
 
     public T pop() {
-        return null;
+        checkStack();
+        T result = (T) stack[size - 1];
+        size--;
```

Comment:
it will also be good to make the element on `stack[size - 1]` null before we decrement `size`